### PR TITLE
bluez: build missing bluetooth utilities

### PR DESCRIPTION
**Pull request by tekeri**
###### Motivation for this change
Some bluetooth developer tools namely hid2hci, rfcomm, sdptool, etc are not built by default.
Some of them do not work out of the box (sdptool requires --compat) but the others seem just work.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/bluez/default.nix [hunk 1]
```diff
@@ -73,10 +73,16 @@ in stdenv.mkDerivation rec {
...
     "--enable-health"
     "--enable-mesh"
-    "--enable-midi"
     "--enable-nfc"
     "--enable-sap"
     "--enable-sixaxis"
+    "--enable-midi"
+    "--enable-btpclient"
+    "--enable-experimental"
+    "--enable-deprecated"
```
Comment (bjornfor) [inline pkgs/os-specific/linux/bluez/default.nix:82]: I'm not familiar with the upstream project, but asking for experimental and deprecated features in the nixpkgs package by default doesn't sound that good to me.

Comment (bjornfor) [reply pkgs/os-specific/linux/bluez/default.nix:82]: >My motivation is to open unix pipe over the air using serial port profile and it seems currently there is no easy alternative tool (except --enable-deprecated).

Well that's... frustrating. (Is there an upstream issue?). What about `--enable-experimental`? Does it really belong here?

Comment (tekeri) [reply pkgs/os-specific/linux/bluez/default.nix:82]: `--enable-experimental` does not strictly belong here. It should be removed, please wait.

Comment (bjornfor) [reply pkgs/os-specific/linux/bluez/default.nix:81]: Maybe you can add a comment about the need for `--enable-deprecated` in the code?

Comment (tekeri) [reply pkgs/os-specific/linux/bluez/default.nix:81]: also removed `--enable-admin` for making sure since it provides a plugin to the daemon.

**Review by bjornfor — APPROVED** [review]

**Comment by tekeri** [comment]
Arch wiki says they have separate packages namely `bluez-utils-compat`.
It should have an additional output, though it can be another PR.

**Pull request merged**